pkg/kubelet/dockertools/manager.go

```diff
@@ -375,7 +375,10 @@ func (dm *DockerManager) GetPodStatus(pod *api.Pod) (*api.PodStatus, error) {
 	for _, status := range pod.Status.ContainerStatuses {
 		oldStatuses[status.Name] = status
 		if status.LastTerminationState.Terminated != nil {
-			lastObservedTime[status.Name] = status.LastTerminationState.Terminated.FinishedAt
+			timestamp, ok := lastObservedTime[status.Name]
+			if !ok || (ok && timestamp.Before(status.LastTerminationState.Terminated.FinishedAt)) {
```

Comment:
What triggers this issue? Why are you introducing this check? I see the correctness, but I want to understand the motivation.